src/spectrecoff-cli/commands/Progress.fs

```diff
@@ -50,6 +50,37 @@ type ProgressDocumentation() =
         setDocumentationStyle
         Many [
             spectreDocSynopsis "Progress module" "This module provides functionality from the progress widget of Spectre.Console" "widgets/progress"
-            docMissing
+            C "A progress can be started by calling the"; P "start"; C"or"; P"startCustom"; C"functions:"
+            funcsOutput [{ Name = "start"; Signature = "(ProgressContext -> Task<unit>) -> Task<unit>" }; { Name = "startCustom"; Signature = "ProgressTemplate -> (ProgressContext -> Task<unit>) -> Task<unit>" }]
+
+            C "The"; P "ProgressTemplate"; C "record can be used to further define the behavior of the progress widget:"
+            propsOutput [
+                { Name = "AutoClear"; Type = (nameof bool); Explanation = "defines if the console should be cleared before progress starts" }
+                { Name = "AutoRefresh"; Type = (nameof bool); Explanation = "defines if the progress should be refreshed automatically" }
+                { Name = "HideCompleted"; Type = (nameof bool); Explanation = "defines if finished tasks should be hidden" }
+                { Name = "Columns"; Type = $"{(nameof ProgressColumn)} list"; Explanation = "defines the columns of the progress widget" }
+            ]
+
+            C "For convenience the"; P "defaultTemplate"; C "and the"; P "emptyTemplate"; C "variables can be used and customized using these functions:"
+            funcsOutput [
+                { Name = "withDescriptionColumn"; Signature = "ProgressTemplate -> ProgressTemplate" }
+                { Name = "withSpinnerColumn"; Signature = "ProgressTemplate -> ProgressTemplate" };
+                { Name = "withProgressBarColumn"; Signature = "ProgressTemplate -> ProgressTemplate" }
+                { Name = "withPercentageColumn"; Signature = "ProgressTemplate -> ProgressTemplate" };
+                { Name = "withRemainingTimeColumn"; Signature = "ProgressTemplate -> ProgressTemplate";  }]
+
+            C "Inside the passed operation, tasks can be added to the"; P "ProgressContext"; C "using the"; P "realizeIn"; C "function:"
```

Comment:
Hmm, I still have trouble understanding this. Do you mean 'past operation', i.e., the operation that has been completed? Or 'passed', as passed from A to B, but then, what are A and B? And what is an _operation_, a _task_?

Reply:
I just saw the type aliases called `StatusOperation `and `ProgressOperation` - do you mean those? Maybe explain them in the doc, and use them in the signatures of the functions?

Reply:
haha, the parameter that is passed, meaning the _function_ that is executed inside that progress widget, is called `operation`. 

After your suggestion, the type alias also has that name, so I hope it is clearer now since I call it like that. If not, we can still revisit :)